src/gwc/src/test/java/org/geoserver/gwc/layer/CatalogConfigurationTest.java

```diff
@@ -407,4 +424,43 @@ private void testModifyLayer(GeoServerTileLayer orig) {
         
         verify(this.tileLayerCatalog, never()).save(info);
     }
+    
+    @Test
+    public void testConfigurationDeadlock() throws Exception {
+        // to make it reproducible with some reliability on my machine
+        // 100000 loops need to be attempted. With the fix it works, but runs for 
+        // a minute and a half, so not suitable for actual builds.
+        // For in-build tests I've thus settled down for 1000 loops instead 
+        final int LOOPS = 1000;
```

Comment:
Do not worry, with 1000 loops it takes 0.7 seconds on a old machine, not really intensive.
But you are right, there is not such a thing as a "slow test" classification in GeoServer